rfc/graphql-schema/additionalTypeDefs/byPage/accountSettings.ts

```diff
@@ -0,0 +1,26 @@
+export default /* GraphQL */ `
+  type CompanyRelationship {
+    company: Company!
+    role: Role!
+  }
```

Comment:
thoughts on this pattern? this scopes role to the customer-company relationship. Open to other suggestions.

The alternatives I thought of were:

```gql
type Customer {
  company: Company # This is backwards incompatible, as we already have a company property
  role: Role # This must be optional, some customers are not b2b, therefore have no role.
  # Also, not sure if it is a common pattern right now, but role might be dependent on the current active company? 🤷 
}
```

bit weirder:

```gql
type Company {
  role: Role! # this is also problematic, only company objects under `customer` will/could include a role.
}

type Customer {
  company: Company # suffers the same issue as above, company exists already
}
```

Reply:
I think the schema you defined w/ it being required fields make sense to me the most.